cli/run_util_pipeline_test.go

```diff
@@ -206,6 +238,21 @@ func TestValidation(t *testing.T) {
 			config:  workflowVariantHasTheSameNameAsAnExistingWorkflowForDAGPipeline,
 			wantErr: "failed to get Bitrise config (bitrise.yml) from base 64 data: Failed to parse bitrise config, error: workflow (b) defined in pipeline (dag) is a variant of another workflow, but it is also defined as a workflow",
 		},
+		{
+			name:    "Generated parallel workflow variant has the same name as an existing workflow",
+			config:  generatedParallelWorkflowVariantHasTheSameNameAsAnExistingWorkflowForDAGPipeline,
+			wantErr: "failed to get Bitrise config (bitrise.yml) from base 64 data: Failed to parse bitrise config, error: parallel workflow variant (a_3) would be generated by workflow (a) defined in pipeline (dag), but it is also defined as a workflow",
+		},
+		{
+			name:    "Parallel value is non-positive", // We let the Pipeline Service define the upper limit for the parallel value
+			config:  parallelValueIsNonPositive,
+			wantErr: "failed to get Bitrise config (bitrise.yml) from base 64 data: Failed to parse bitrise config, error: workflow (a) defined in pipeline (dag) has invalid parallel value (0), should be at least 1",
+		},
+		{
+			name:    "Parallel value is non-integer",
+			config:  parallelValueIsNonInteger,
+			wantErr: "failed to get Bitrise config (bitrise.yml) from base 64 data: Failed to parse bitrise config, error: yaml: unmarshal errors:\n  line 6: cannot unmarshal !!bool `true` into int",
```

Comment:
I understand we didn't "do" this, but it felt valuable to document the behaviour. (I suppose we don't want to introduce custom unmarshaling to be able to provide a nicer error message here.)

Reply:
Actually the upcoming dynamic parallel feature will make this a bit more complicated as this field has to be of a string type to store the env var. 

But even then I think we do not need custom unmarshalling just extra post processing (like checking if the string is either an int or starts with the dollar sign) during validation.